### PR TITLE
Remove usage of the NODE_ADMIN global in RTDB

### PR DESCRIPTION
**Pull request by samtstern**
### Discussion

The `NODE_ADMIN` global means that the RTDB Admin and Client SDKs cannot coexist, whichever is initialized first will control the behavior.

This is a problem for testing, where clients want to use the Admin SDK to do privileged actions but then test the behavior of the Client SDK with low privileges.

### Testing

- [x] New unit test in `rules-unit-testing`

### API Changes

N/A

### CODE REVIEW

File: packages/database/src/realtime/WebSocketConnection.ts
Comment (yuchenshi) [inline packages/database/src/realtime/WebSocketConnection.ts:90]: I don't think we need a full reference to repoInfo. Can we just grab `nodeAdmin` as a field?

**Review by yuchenshi — APPROVED** [review]

File: packages/database/src/realtime/WebSocketConnection.ts
Comment (samtstern) [reply packages/database/src/realtime/WebSocketConnection.ts]: You caught me being lazy

File: packages/database/index.node.ts
Comment (Feiyang1) [inline packages/database/index.node.ts:66]: We can remove this constant since RTDB is the only consumer of it.  If any other JS SDKs were used by the admin SDK in the future, they should not rely on this constant either.

Comment (yuchenshi) [reply packages/database/index.node.ts:66]: We should hold off on this. `firebase-admin` does not pin its dependency against `@firebase/database`, so we can only remove this in a major version bump to avoid breaking customers.

Comment (yuchenshi) [reply packages/database/index.node.ts:66]: Oh I think I misunderstood -- this is all internal in this package and utils

Comment (samtstern) [reply packages/database/index.node.ts:66]: @Feiyang1 so I still want `isNodeSdk()` to work ... where does `NODE_ADMIN` actually come from when it's not set in `initStandalone`?  Is it set in part of the build process?

Comment (Feiyang1) [reply packages/database/index.node.ts:66]: It defaults to `false` unless set by `initStandalone`. https://github.com/firebase/firebase-js-sdk/blob/master/packages/util/src/constants.ts#L30

Comment (samtstern) [reply packages/database/index.node.ts:66]: I'm not comfortable removing this at the moment ... it's used within `isNodeSdk()` which is used in many places including one place where `module.exports` is redefined.  I think removing this is likely to break something I don't understand.  Maybe someone from JSCore feels confident enough to do it as a follow up.

Comment (Feiyang1) [reply packages/database/index.node.ts:66]: The check for CONSTANTS.NODE_ADMIN is unnecessary in `isNodeSdk()` because `CONSTANTS.NODE_CLIENT` will be always true when `CONSTANTS.NODE_ADMIN` is true. Sure, I will do a follow up PR for this.

**Comment by samtstern** [comment]
@schmidt-sebastian I need your codeowners review on this!

**Review by schmidt-sebastian — APPROVED** [review]
Basically LGTM, though I wonder if we can simplify this a bit: https://gist.github.com/schmidt-sebastian/08c74a0160809165fed81e31088cf8ce

If not, we can merge as is.

**Comment by samtstern** [comment]
@hsubox76 says I can merge over this error, so merging!

**Pull request merged**
